### DISCUSSION TITLE
How do you get QField "control" slot props

### DISCUSSION THREAD

**Discussion by dinyo** [quasarframework/quasar#16450] [General - Components / Directives / etc] [answered]
sorry i am noob at this, but how do you access props from "control" slot in qfield component? i need the focused state, but the qfield component itself not sharing the state. thanks.

example:
https://codepen.io/dinyo/pen/VwqRaJM

**Comment by yusufkandemir** [comment]
`props.focused`

**Reply by dinyo** [reply]
no, i mean outside the <template #control>

**Reply by yusufkandemir** [reply]
Why do you need it? How will you use it?

**Reply by dinyo** [reply]
for doing something in <script setup>

```
<template>
      <q-field outlined dense>
        <template #control="ctrlProps">
          <div>some text...</div>
        </template>
      </q-field>
</template>

<script setup>
//access ctrlProps here to do something
</script>
```

**Comment by metalsadman** [comment, marked as answer]
you can't access it directly from template to script, you can use some handler, let's say on your div you have a click event and pass the props as parameter. like so:
https://codepen.io/metalsadman/pen/mdaZqyr

or use something like https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver or https://vueuse.org/core/useMutationObserver/

or Quasar's https://quasar.dev/vue-directives/mutation#catch-everything --> this is easier to use imo.

**Reply by dinyo** [reply]
thanks!
